tests/test_api_v1.py

```diff
@@ -1130,24 +1130,46 @@ def test_set_state_stale_failed_to_schedule(
     mock_cleanup_npm.delay.assert_called_once()
 
 
-def test_set_state_from_stale(app, client, db, worker_auth_env):
+@pytest.mark.parametrize(
+    "initial_state,to_state,expected_resp_code",
+    [
+        ["in_progress", "in_progress", HTTPStatus.OK],
+        ["in_progress", "complete", HTTPStatus.OK],
+        ["in_progress", "failed", HTTPStatus.OK],
+        ["in_progress", "stale", HTTPStatus.OK],
+        ["complete", "stale", HTTPStatus.OK],
+        ["failed", "stale", HTTPStatus.OK],
+        # Invalid transition
+        ["complete", "in_progress", HTTPStatus.BAD_REQUEST],
+        ["failed", "in_progress", HTTPStatus.BAD_REQUEST],
+        ["stale", "in_progress", HTTPStatus.BAD_REQUEST],
+        ["stale", "complete", HTTPStatus.BAD_REQUEST],
+        ["stale", "failed", HTTPStatus.BAD_REQUEST],
```

Comment:
Thank you for your work! Maybe it worths adding these invalid transactions 
![image](https://user-images.githubusercontent.com/25501330/118448828-c1851f80-b6f2-11eb-825f-7724b19f8ce2.png)

Reply:
@akhmanova that's a pretty cool graphic, how did you make it?

Reply:
@chmeliik I started to use miro.com instead of paper notes. Because
- paper notes usually are the mess
- it is easy to share something virtual

maybe it is eco friendly as well :D But I don't know 100%